pandas/core/groupby/generic.py

```diff
@@ -1863,15 +1864,138 @@ def _transform_general(self, func, engine, engine_kwargs, *args, **kwargs):
     3  5  9
     4  5  8
     5  5  9
+
+    Using list-like arguments
+
+    >>> df = pd.DataFrame({"col": list("aab"), "val": range(3)})
+    >>> df.groupby("col").transform(["sum", "min"])
+        val
+       sum  min
+    0    1    0
+    1    1    0
+    2    2    2
+
+    .. versionchanged:: 3.0.0
+
+    Named aggregation
+
+    >>> df = pd.DataFrame({"A": list("aaabbbccc"), "B": range(9), "D": range(9, 18)})
+    >>> df.groupby("A").transform(
+    ...     b_min=pd.NamedAgg(column="B", aggfunc="min"),
+    ...     c_sum=pd.NamedAgg(column="D", aggfunc="sum")
+    ... )
+       b_min  c_sum
+    0      0     30
+    1      0     30
+    2      0     30
+    3      3     39
+    4      3     39
+    5      3     39
+    6      6     48
+    7      6     48
+    8      6     48
+
+    .. versionchanged:: 3.0.0
     """
     )
 
     @Substitution(klass="DataFrame", example=__examples_dataframe_doc)
     @Appender(_transform_template)
-    def transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
-        return self._transform(
-            func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+    def transform(
+        self,
+        func: None
+        | (Callable | str | list[Callable | str] | dict[str, NamedAgg]) = None,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> DataFrame:
+        if func is None:
+            # Convert named aggregations to dictionary format
+            transformed_func = {
+                name: aggfunc
+                for name, aggfunc in kwargs.items()
+                if not isinstance(aggfunc[1], DataFrame)
+            }
+            kwargs = {}
+            return self._transform_multiple_funcs(
+                transformed_func,
+                *args,
+                engine=engine,
+                engine_kwargs=engine_kwargs,
+                **kwargs,
+            )
+        else:
+            if isinstance(func, list):
+                func = maybe_mangle_lambdas(func)
+                return self._transform_multiple_funcs(
+                    func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+                )
+            else:
+                return self._transform(
+                    func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+                )
+
+    def _transform_multiple_funcs(
+        self,
+        func: Any,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> DataFrame:
+        results = []
+        if isinstance(func, dict):
+            for name, named_agg in func.items():
+                column_name = named_agg.column
+                agg_func = named_agg.aggfunc
+                result = self._transform_single_column(
```

Comment:
Just to clarify @rhshadrach 
that would do the same thing as `.transform(["sum", "min"])`?
but with columns names "a" and "b" instead of "sum" and "min"?

Reply:
No, `.transform(["sum", "min"])` would also act on other columns if there are any; `.transform({"a": "sum", "b": "mean"})` will only act on columns `a` and `b`.

Reply:
Understood, added the functionality including tests.
Also some changes dealing with list-like.
The behavior when used with more columns was wrong:
  val another val other
  sum     sum min     min
0   1       1   0       0
1   1       1   0       0
2   2       2   2       2

was doing that instead of:
  val     another    
  sum min     sum min
0   1   0       1   0
1   1   0       1   0
2   2   2       2   2

It's ok now, added also to the tests